java/ql/src/experimental/CWE-532/SensitiveInfoLog.ql

```diff
@@ -0,0 +1,83 @@
+/**
+ * @id java/sensitiveinfo-in-logfile
+ * @name Insertion of sensitive information into log files
+ * @description Writting sensitive information to log files can give valuable guidance to an attacker or expose sensitive user information.
+ * @kind problem
+ * @tags security
+ *       external/cwe-532
+ */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import DataFlow
+import PathGraph
+
+/** Class of popular logging utilities **/
+class LoggerType extends RefType {
+    LoggerType() {
+        this.hasQualifiedName("org.apache.log4j", "Category") or //Log4J
+        this.hasQualifiedName("org.slf4j", "Logger")  //SLF4j
+    }
```

Comment:
Please add the Gradle Logger:
https://docs.gradle.org/current/javadoc/org/gradle/api/logging/Logger.html

We actually just dealt with an issue like this.
https://blog.gradle.org/plugin-portal-update

Reply:
Thanks @JLLeitschuh again for this recommendation.

However, as shown in [Gradle Interface Logger](https://docs.gradle.org/current/javadoc/org/gradle/api/logging/Logger.html) referenced in [Gradle Logging Documentation](https://docs.gradle.org/current/userguide/logging.html), org.gradle.api.logging.Logger already extends org.slf4j.Logger thus we don't need to explicitly add the following statement:
```
this.hasQualifiedName("org.gradle.api.logging", "Logger")  //Gradle Logging
```

I will change the comment on the following line
```
this.hasQualifiedName("org.slf4j", "Logger") //SLF4j
```
to
```
this.hasQualifiedName("org.slf4j", "Logger") //SLF4j and Gradle Logging
```

Please let me know if I miss something. Thanks.